clientv3/balancer_test.go

```diff
@@ -237,3 +238,82 @@ func (kcl *killConnListener) close() {
 	close(kcl.stopc)
 	kcl.wg.Wait()
 }
+
+func TestBalancerTrySwitch(t *testing.T) {
```

Comment:
The mock grpc notifications this test uses could diverge from grpc behavior and the test itself is rather complicated. A clientv3/integration/kv_test.go test would be better instead since what matters is the Get failing over:

```go
func TestKVGetFailoverPartition(t *testing.T) {
        defer testutil.AfterTest(t)
        clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
        defer clus.Terminate(t)

        cli, err := clientv3.New(clientv3.Config{
                Endpoints:   clus.Client(0).Endpoints(),
                DialTimeout: time.Second,
        })
        testutil.AssertNil(t, err)
        defer cli.Close()

        // force to connect to member 0
        _, err = cli.Get(context.TODO(), "abc")
        testutil.AssertNil(t, err)

        // permit failover to 1
        cli.SetEndpoints(append(cli.Endpoints(), clus.Client(1).Endpoints()...)...)

        // partition member 0 from rest of cluster
        clus.Members[0].InjectPartition(t, clus.Members[1:])

        // may have triggered a leader election; expect timeouts
        timeout := clus.Members[1].ServerConfig.ReqTimeout()
        // member 0 fails to get consensus for the l-read, balancer switches
        ctx, cancel := context.WithTimeout(context.TODO(), 3*timeout)
        _, err = cli.Get(ctx, "abc")
        cancel()
        testutil.AssertNil(t, err)
}
```